lib/access_import/client_transform.rb

```diff
@@ -0,0 +1,127 @@
+      relatives_attributes: relatives_attrs(relatives),
+      access_import: access_import(begleitet, haupt_person, personen_rolle,
+        familien_rolle, land, sprachen)
+    }
+  end
+
+  def comments(begleitet, personen_rolle, haupt_person)
+    comments = ''
+    comments += "#{begleitet[:m_Bemerkung]}\n\n" if begleitet[:m_Bemerkung]
+    comments += "#{personen_rolle[:m_Bemerkungen]}\n\n" if personen_rolle[:m_Bemerkungen]
+    comments += "#{haupt_person[:m_Bemerkungen]}\n\n" if haupt_person[:m_Bemerkungen]
+    comments
+  end
+
+  def handle_begleitete(personen_rolle, haupt_person)
+    begleitete = @begleitete.where_personen_rolle(personen_rolle[:pk_PersonenRolle])
+    return [begleitete.first[1], []] if begleitete.size == 1
+    begleitet = begleitete.select do |_key, beg|
+      beg[:fk_FamilienRolle] == 2 # Hauptperson
+    end
+    if begleitet.size == 1
+      return [begleitet.first[1], begleitete.except(begleitet.first[0])]
+    end
+    begleitet = begleitete.select do |_key, beg|
+      beg[:t_Vorname] == haupt_person[:t_Vorname]
+    end
+    if begleitet.size == 1
+      return [begleitet.first[1], begleitete.except(begleitet.first[0])]
+    end
+    begleitet = begleitete.select do |_key, beg|
+      haupt_person[:t_Vorname].include? beg[:t_Vorname]
+    end
+    if begleitet.size.positive?
+      return [begleitet.first[1], begleitete.except(begleitet.first[0])]
+    end
+    # nothing worked, just take first
+    [begleitet.first[1], begleitete.except(begleitet.first[0])]
+  end
+
+  def relatives_attrs(relatives)
+    relatives.map do |_key, relative|
+      [(Time.now.to_f * 1000).to_i, {
+        first_name: relative[:t_Vorname],
+        last_name: relative[:t_Name], relation: relative[:relation],
+        birth_year: relative[:z_Jahrgang] && Date.parse("#{relative[:z_Jahrgang]}-01-01")
+      }]
+    end.to_h
+  end
+
+  def birth_year(geburtsdatum, jahrgang)
+    return geburtsdatum if geburtsdatum
+    Date.parse(jahrgang + '-06-01') if jahrgang
+  end
+
+  def language_skills_attributes(sprachen)
+    sprachen.map do |sprache|
+      [
+        (Time.now.to_f * 1000).to_i,
+        { language: sprache[:language][:lang], level: sprache[:kenntnisstufe_ve] }
+      ]
+    end.to_h
+  end
+
+  def access_import(*fields)
+    {
+      id_personen_rolle: fields[2][:pk_PersonenRolle],
+      begleitet: down_hkeys(fields[0].slice(:fk_FamilienRolle, :fk_PersonenRolle, :pk_Begleitete,
+        :m_Bemerkung, :z_Jahrgang)),
+      haupt_person: down_hkeys(fields[1].slice(:b_KlientAOZ, :fk_Land, :fk_PLZ, :t_Anrede,
+        :t_NNummer, :pk_Hauptperson)),
+      personen_rolle: down_hkeys(fields[2]),
+      familien_rolle: down_hkeys(fields[3].except(:d_MutDatum, :t_Mutation)),
+      land: fields[4]
+    }
+  end
+
+  def contact_attributes(haupt_person, plz)
+    mobile_number(haupt_person).merge(
+      first_name: haupt_person[:t_Vorname] || 'unbekannt',
+      last_name: haupt_person[:t_Nachname] || 'unbekannt',
```

Comment:
Not sure how empty strings should be handled, you could do `haupt_person[:t_Nachname].presence || 'unbekannt'` here (it's similar to `present?` and returns `nil` for blank strings)